enterprise/ssl-impl/src/main/java/io/crate/protocols/ssl/SslContextProviderService.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * This file is part of a module with proprietary Enterprise Features.
+ *
+ * Licensed to Crate.io Inc. ("Crate.io") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Unauthorized copying of this file, via any medium is strictly prohibited.
+ *
+ * To use this file, Crate.io must have given you permission to enable and
+ * use such Enterprise Features and you must have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  If you enable or use the Enterprise
+ * Features, you represent and warrant that you have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  Your use of the Enterprise Features
+ * if governed by the terms and conditions of your Enterprise or Subscription
+ * Agreement with Crate.io.
+ */
+
+package io.crate.protocols.ssl;
+
+import org.elasticsearch.common.component.AbstractLifecycleComponent;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.threadpool.Scheduler;
+import org.elasticsearch.threadpool.ThreadPool;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+
+import static java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY;
+
+@Singleton
+public class SslContextProviderService extends AbstractLifecycleComponent {
+
+    private static final TimeValue FILES_WATCH_POLL_INTERVAL = TimeValue.timeValueSeconds(5);
+
+    private final SslContextProvider sslContextProvider;
+    private final Settings settings;
+    private final ThreadPool threadPool;
+
+    private volatile FilesWatcher filesWatcher;
```

Comment:
`doStart` / `doStop` is afaik always called in the same thread, so `volatile` is not necessary here.